lib/collection/request.js

```diff
@@ -251,6 +311,50 @@ _.assign(Request.prototype, /** @lends Request.prototype */ {
         this.url.removeQueryParams(params);
     },
 
+    /**
+     * Get the request size by computing the headers and body or using the
+     * actual content length header once the request is sent.
+     *
+     * @returns {Object}
+     */
+    size: function () {
+        var contentLength = this.headers.get(CONTENT_LENGTH),
+            requestTarget = this.url.getPathWithQuery(),
+            bodyString,
+            sizeInfo = {
+                body: 0,
+                header: 0,
+                total: 0
```

Comment:
Need to add the “source” section to indicate whether this is from content length or computer.